tcpclient/connectionsManagement.go

```diff
@@ -44,15 +48,27 @@ func TCPConnect(id int, host string, port int, wg *sync.WaitGroup, debugOut io.W
 	reportConnectionStatus(debugOut, statusChannel, connectionDescription)
 	connBuf := bufio.NewReader(conn)
 	for {
-		str, err := connBuf.ReadString('\n')
-		if len(str) > 0 {
-			fmt.Fprintln(debugOut, "Connection", id, "got", str)
-		}
-		if err != nil {
-			connectionDescription.status = ConnectionClosed
-			reportConnectionStatus(debugOut, statusChannel, connectionDescription)
-			wg.Done()
-			return err
+		select {
+		case <-closeRequest:
+			// we need to close gracefully the connection
+		default:
+			// TODO: to review
+			conn.SetReadDeadline(time.Now().Add(time.Duration(defaultReadTimeoutInSecs) * time.Second))
+			str, err := connBuf.ReadString('\n')
+			if len(str) > 0 {
+				fmt.Fprintln(debugOut, "Connection", id, "got", str)
+			}
+			if err, ok := err.(net.Error); ok && err.Timeout() {
+				fmt.Fprintln(debugOut, "Connection", id, "timed out reading. Trying again..")
+			} else if err != nil {
+				fmt.Fprintln(debugOut, "Connection", id, "looks closed. Error", reflect.TypeOf(err),"when reading:")
```

Comment:
first time I see use of `reflect` package, something new to learn :)

Reply:
:+1: